test_run = require('test_run').new()
---
...
engine = test_run:get_cfg('engine')
---
...
_ = box.space._session_settings:update('sql_default_engine', {{'=', 2, engine}})
---
...
box.cfg{}
---
...
box.execute("CREATE TABLE t1 (s1 INTEGER PRIMARY KEY AUTOINCREMENT, s2 INTEGER, CHECK (s1 <> 19));");
---
- row_count: 1
...
box.execute("CREATE TABLE t2 (s1 INTEGER PRIMARY KEY AUTOINCREMENT, s2 INTEGER, CHECK (s1 <> 19 AND s1 <> 25));");
---
- row_count: 1
...
box.execute("CREATE TABLE t3 (s1 INTEGER PRIMARY KEY AUTOINCREMENT, s2 INTEGER, CHECK (s1 < 10));");
---
- row_count: 1
...
box.execute("insert into t1 values (18, null);")
---
- row_count: 1
...
box.execute("insert into t1(s2) values (null);")
---
- null
- Check constraint 'ck_unnamed_T1_1' failed for a tuple
...
box.execute("insert into t2 values (18, null);")
---
- row_count: 1
...
box.execute("insert into t2(s2) values (null);")
---
- null
- Check constraint 'ck_unnamed_T2_1' failed for a tuple
...
box.execute("insert into t2 values (24, null);")
---
- row_count: 1
...
box.execute("insert into t2(s2) values (null);")
---
- null
- Check constraint 'ck_unnamed_T2_1' failed for a tuple
...
box.execute("insert into t3 values (9, null)")
---
- row_count: 1
...
box.execute("insert into t3(s2) values (null)")
---
- null
- Check constraint 'ck_unnamed_T3_1' failed for a tuple
...
box.execute("DROP TABLE t1")
---
- row_count: 1
...
box.execute("DROP TABLE t2")
---
- row_count: 1
...
box.execute("DROP TABLE t3")
---
- row_count: 1
...
box.func.check_T1_ck_unnamed_T1_1:drop()
---
...
box.func.check_T2_ck_unnamed_T2_1:drop()
---
...
box.func.check_T3_ck_unnamed_T3_1:drop()
---
...
